statistics/handle/handle_hist.go

```diff
@@ -289,34 +324,66 @@ func (h *Handle) readStatsForOne(col model.TableColumnID, c *statistics.Column,
 	return colHist, nil
 }
 
+func (h *Handle) readStatsForOneIdx(idx model.TableItemID, index *statistics.Index, reader *statsReader) (*statistics.Index, error) {
+	loadFMSketch := config.GetGlobalConfig().Performance.EnableLoadFMSketch
+	hg, err := h.histogramFromStorage(reader, idx.TableID, idx.ID, types.NewFieldType(mysql.TypeBlob), index.Histogram.NDV, 1, index.LastUpdateVersion, index.NullCount, index.TotColSize, index.Correlation)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	cms, topN, err := h.cmSketchAndTopNFromStorage(reader, idx.TableID, 1, idx.ID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	var fms *statistics.FMSketch
+	if loadFMSketch {
+		fms, err = h.fmSketchFromStorage(reader, idx.TableID, 1, idx.ID)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	rows, _, err := reader.read("select stats_ver from mysql.stats_histograms where is_index = 0 and table_id = %? and hist_id = %?", idx.TableID, idx.ID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if len(rows) == 0 {
+		logutil.BgLogger().Error("fail to get stats version for this histogram", zap.Int64("table_id", idx.TableID), zap.Int64("hist_id", idx.ID))
```

Comment:
Ditto. And maybe we should use `rows[0].GetInt64(0)` as `StatsVer`, otherwise `select stats_ver` is unused.